test/testdrive/kafka-avro-sinks-doc-comments.td

```diff
@@ -0,0 +1,106 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test Avro UPSERT sinks doc comments
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_comment = true;
+
+> CREATE TYPE point AS (x integer, y integer);
+> CREATE TYPE custom_map AS MAP (KEY TYPE = text, VALUE TYPE = bool)
+> CREATE TABLE t (c1 point, c2 text NOT NULL, c3 custom_map, c4 point list);
+> INSERT INTO t SELECT ROW(1, 1)::point AS c1, 'text' AS c2, '{a=>true}'::custom_map as c3, LIST[ROW(1, 1)::point] as c4;
+
+> COMMENT ON TABLE t IS 'comment on table t';
+> COMMENT ON COLUMN t.c3 IS 'comment on column t.c3';
+> COMMENT ON COLUMN t.c4 IS 'comment on column t.c4';
+> COMMENT ON TYPE point IS 'comment on type point';
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE CONNECTION IF NOT EXISTS csr_conn TO CONFLUENT SCHEMA REGISTRY (
+    URL '${testdrive.schema-registry-url}'
+  );
+
+> CREATE SINK sink1 FROM t
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-sink1-${testdrive.seed}')
+  KEY (c2) NOT ENFORCED
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+  (
+   DOC ON COLUMN t.c1 = 'doc on t.c1',
```

Comment:
@def-  given the extra syntax that is being added, I think a platform check would be nice to have here. Or, just have the existing sinks.py put some comments and then use `SHOW CREATE` to check for them in `validate()`

Reply:
Yes, already on my list.